src/org/zaproxy/zap/extension/alert/AlertViewPanel.java

```diff
@@ -291,134 +313,61 @@ public void keyPressed(java.awt.event.KeyEvent evt) {
-						LayoutHelper.getGBC(0, gbcRow, 1, 0.25D, new Insets(1,1,1,1)));
-				alertDisplay.add(alertCweId, 
-						LayoutHelper.getGBC(1, gbcRow, 1, 0.75D, new Insets(1,1,1,1)));
-				gbcRow++;
-				
-				alertDisplay.add(new JLabel(Constant.messages.getString("alert.label.wascid")),
-						LayoutHelper.getGBC(0, gbcRow, 1, 0.25D, new Insets(1,1,1,1)));
-				alertDisplay.add(alertWascId, 
-						LayoutHelper.getGBC(1, gbcRow, 1, 0.75D, new Insets(1,1,1,1)));
-				gbcRow++;
-
-				alertDisplay.add(new JLabel(Constant.messages.getString("alert.label.source")),
-						LayoutHelper.getGBC(0, gbcRow, 1, 0.25D, new Insets(1,1,1,1)));
-				alertDisplay.add(alertSource, 
-						LayoutHelper.getGBC(1, gbcRow, 1, 0.75D, new Insets(1,1,1,1)));
-				gbcRow++;
-			}
+			alertUrl.setEditable(false);
+
+			alertDisplay.add(editable ? alertEditName : alertName,
+					LayoutHelper.getGBC(0, gbcRow, 2, 0, DEFAULT_INSETS));
+			// Show a blank label instead of the edit button if already editing
+			gbcRow++;
+			alertDisplay.add(getUrlLabel(), LayoutHelper.getGBC(0, gbcRow, 1, 0, DEFAULT_INSETS));
+			alertDisplay.add(alertUrl, LayoutHelper.getGBC(1, gbcRow, 1, 1, DEFAULT_INSETS));
+			gbcRow++;
+			alertDisplay.add(getRiskLabel(), LayoutHelper.getGBC(0, gbcRow, 1, 0, DEFAULT_INSETS));
+			alertDisplay.add(editable ? alertEditRisk : alertRisk,
+					LayoutHelper.getGBC(1, gbcRow, 1, 1, DEFAULT_INSETS));
+			gbcRow++;
+			alertDisplay.add(getConfidenceLabel(), LayoutHelper.getGBC(0, gbcRow, 1, 0, DEFAULT_INSETS));
+			alertDisplay.add(editable ? alertEditConfidence : alertConfidence,
+					LayoutHelper.getGBC(1, gbcRow, 1, 1, DEFAULT_INSETS));
+			gbcRow++;
+			alertDisplay.add(getParameterLabel(), LayoutHelper.getGBC(0, gbcRow, 1, 0, DEFAULT_INSETS));
+			alertDisplay.add(editable ? alertEditParam : alertParam,
+					LayoutHelper.getGBC(1, gbcRow, 1, 1, DEFAULT_INSETS));
+			gbcRow++;
+			alertDisplay.add(getAttackLabel(), LayoutHelper.getGBC(0, gbcRow, 1, 0, DEFAULT_INSETS));
+			alertDisplay.add(editable ? alertEditAttack : alertAttack,
+					LayoutHelper.getGBC(1, gbcRow, 1, 1, DEFAULT_INSETS));
+			gbcRow++;
+			alertDisplay.add(getEvidenceLabel(), LayoutHelper.getGBC(0, gbcRow, 1, 0, DEFAULT_INSETS));
+			alertDisplay.add(editable ? alertEditEvidence : alertEvidence,
+					LayoutHelper.getGBC(1, gbcRow, 1, 1, DEFAULT_INSETS));
+			gbcRow++;
+			alertDisplay.add(getCweidLabel(), LayoutHelper.getGBC(0, gbcRow, 1, 0, DEFAULT_INSETS));
+			alertDisplay.add(editable ? alertEditCweId : alertCweId,
+					LayoutHelper.getGBC(1, gbcRow, 1, 1, DEFAULT_INSETS));
+			gbcRow++;
+			alertDisplay.add(getWascidLabel(), LayoutHelper.getGBC(0, gbcRow, 1, 0, DEFAULT_INSETS));
+			alertDisplay.add(editable ? alertEditWascId : alertWascId,
+					LayoutHelper.getGBC(1, gbcRow, 1, 1, DEFAULT_INSETS));
+			gbcRow++;
+			alertDisplay.add(getSourceLabel(), LayoutHelper.getGBC(0, gbcRow, 1, 0, DEFAULT_INSETS));

```

Comment:
This should probably be shown only when not editing (otherwise we should set the `alertSource` always to not leave the field empty).